packages/melange.ppx/ppx.ml

```diff
@@ -401,6 +544,20 @@ module Debug = struct
 end
 
 let () =
-  Driver.register_transformation ~impl:structure_mapper
+  Driver.add_arg "-bundler"
+    (String
+       (fun str ->
+         match str with
+         | "webpack" -> Mel_module.bundler := Webpack
+         | "esbuild" -> Mel_module.bundler := Esbuild
+         | _ ->
+             failwith
+               (Printf.sprintf
+                  {|Unknown value %S passed as -bundler flag in melange.ppx, valid values: "webpack", "esbuild"|}
+                  str)))
+    ~doc:
+      "generate paths to assets in mel.module using the file name scheme of \
+       the bundler of choice";
+  Driver.V2.register_transformation ~impl:structure_mapper
```

Comment:
Changing to `V2` here is what allows to get the `ctxt` in `structure_mapper`, which is used later on to get the path to the file being processed (so that we can open it to get its content for the hash).